crates/floresta-node/src/json_rpc/res.rs

```diff
@@ -310,6 +366,15 @@ pub enum JsonRpcError {
 
     /// Raised if when the rescanblockchain command, with the timestamp flag activated, contains some timestamp thats less than the genesis one and not zero which is the default value for this arg.
     InvalidTimestamp,
+
+    /// Raised when we fail to parse a value to a proper json
+    ToValue(serde_json::Error),
+
+    InvalidHeight(Box<dyn std::error::Error + Send + Sync>),
+
+    InvalidBlockHash(Box<dyn std::error::Error + Send + Sync>),
+
+    InvalidHeader(Box<dyn std::error::Error + Send + Sync + 'static>),
```

Comment:
I fell like they are basically `BlockNotFound`. Also, no need to use these `Box`-ed errors